src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -66,22 +76,56 @@
         return getAcceptableTokens();
     }
 
+    /**
+     * Whether to allow default keyword not in last but surrounded with case.
+     * @param newValue whether to ignore checking.
+     */
+    public void setSkipIfLastAndSharedWithCase(boolean newValue) {
+        skipIfLastAndSharedWithCase = newValue;
+    }
+
     @Override
     public void visitToken(DetailAST ast) {
         final DetailAST defaultGroupAST = ast.getParent();
         //default keywords used in annotations too - not what we're
         //interested in
         if (defaultGroupAST.getType() != TokenTypes.ANNOTATION_FIELD_DEF
                 && defaultGroupAST.getType() != TokenTypes.MODIFIERS) {
-            final DetailAST switchAST = defaultGroupAST.getParent();
-            final DetailAST lastGroupAST =
-                switchAST.getLastChild().getPreviousSibling();
 
-            if (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
-                || defaultGroupAST.getColumnNo()
-                    != lastGroupAST.getColumnNo()) {
+            if (skipIfLastAndSharedWithCase) {
+                if (findNextSibling(ast, TokenTypes.LITERAL_CASE) == TokenTypes.LITERAL_CASE) {
+                    log(ast, MSG_KEY_SKIP_IF_LAST_AND_SHARED_WITH_CASE);
+                }
+                else if (ast.getPreviousSibling() == null
+                    && findNextSibling(ast.getParent(),
+                        TokenTypes.CASE_GROUP) == TokenTypes.CASE_GROUP) {
+                    log(ast, MSG_KEY);
+                }
+            }
+            else if (findNextSibling(ast.getParent(),
+                TokenTypes.CASE_GROUP) == TokenTypes.CASE_GROUP) {
                 log(ast, MSG_KEY);
             }
         }
     }
+
+    /**
+     * Return token type only if passed tokenType in argument is found or returns -1.
+     *
+     * @param ast root node.
+     * @param tokenType tokentype to be processed.
+     * @return token id if desired token is not found then -1.
+     */
+    private static int findNextSibling(DetailAST ast, int tokenType) {
+        int token = -1;
```

Comment:
Return from method Details at object that you found , and in place check for not null